server/src/main/java/org/elasticsearch/synonyms/SynonymsManagementAPIService.java

```diff
@@ -112,7 +115,25 @@ private static XContentBuilder mappings() {
         }
     }
 
-    public void getSynonymsSet(String resourceName, int from, int size, ActionListener<SynonymsSetResult> listener) {
+    public void listSynonymsSet(int from, int size, ActionListener<PagedResult<SynonymSetSummary>> listener) {
+        client.prepareSearch(SYNONYMS_ALIAS_NAME)
+            .setSize(0)
+            .addAggregation(new TermsAggregationBuilder(SYNONYM_SETS_AGG_NAME).field(SYNONYMS_SET_FIELD).order(BucketOrder.key(true)))
+            .setPreference(Preference.LOCAL.type())
+            .execute(listener.delegateFailure((searchResponseListener, searchResponse) -> {
+                Terms aggregation = searchResponse.getAggregations().get(SYNONYM_SETS_AGG_NAME);
+                List<? extends Terms.Bucket> buckets = aggregation.getBuckets();
+                SynonymSetSummary[] synonymSetSummaries = buckets.stream()
+                    .skip(from)
+                    .limit(size)
+                    .map(bucket -> new SynonymSetSummary(bucket.getDocCount(), bucket.getKeyAsString()))
+                    .toArray(SynonymSetSummary[]::new);
+
+                listener.onResponse(new PagedResult<>(buckets.size(), synonymSetSummaries));
+            }));
+    }
+
+    public void getSynonymsSet(String resourceName, int from, int size, ActionListener<PagedResult<SynonymRule>> listener) {
```

Comment:
Sorry, may be we should also rename this function to `getSynonymRules` and rename `listSynonymsSet` to `getSynonymsSet`?

Reply:
Makes sense!